### ISSUE TITLE
Some options still "on" when restored to their (perceived) defaults

### ISSUE THREAD

**Issue by lederer** [azavea/cac-tripplanner#805]
The options menu behavior was insufficiently spec'd.

### Desired behavior

Consider the **Bikeshare**, **Ride quality**, and **Accessibility** options. Each should have two meta-states: `off` and `on`. By default each should be `off`. 

When you open an option's sub-menu, there are two or more states to choose from, one of which should be functionally equivalent to `off` (I have a bike, Any ride quality, No wheelchair). The others should be functionally equivalent to `on`.

Choosing an `on` state for any option should result in: (1) that option being highlighted in the options root menu, (2) the green options bar being shown.

If an option is `on`, turning it `off` by manually clicking its off state in its menu should be functionally equivalent to it never having been turned on in the first place. That is, (1) the option should no longer be highlighted in the options root menu, and (2) assuming no other options are currently `on`, the green options bar should be visible.

### Current behavior

Instead the currently logic is that each option is `off` by default, but if you turn it `on` then subsequently click its `off` state, it is still considered `on` because you manually chose this state.

Repro:
1. Go to map view. Refresh if necessary to clear any current options.
1. Options > {Bikeshare, Ride quality, Accessibility}
1. Set Bikeshare to Indego, or Ride quality to !Any, or Accessibility to Wheelchair.
   - Item is highlighted as "on" in the options root menu.
1. Click Close.
   - Green options bar appears.
1. Options > Same option as before.
1. Click the default option.
   - Item is still highlighted in the options root menu.
   - EXPECTED the option to appear "off", as if it had never been set.
1. Click Close.
   - Green options bar still visible.
   - EXPECTED it to disappear.

### Arrival/Departure timing

This option behaves as desired. It starts as `off`. It can be turned `on` by setting its value. It can be turned `off` by clicking `Clear`.  

This issue is effectively asking that a click on any of the the other options' "default" state should be like clicking `Clear`.

**Comment by flibbertigibbet** [comment]
Clicking an item selects; it does not toggle. This is intentional behavior. The only option which may be cleared is the date/time, which has an explicit 'clear' button.

Requested behavior would add undue complexity for unclear benefit.

**Comment by lederer** [comment]
Right, that's the current behavior. But it's a fundamentally untenable user experience. 

As it stands, each of those options has 2 semantic states but behaves visually as if there are 3 states — even though 2 of them are semantically and functionally the same.[*]

Consider this flow… When the app starts fresh, if you open walking options, Accessibility is "off", which semantically and functionally means "No wheelchair". Indeed if you open Accessibility, you see a menu where the option for "No wheelchair" is visually selected, meaning it is the default state (ie, it is the selected state even though you haven't manually selected a state). If you change it to "I have a wheelchair" and then change it back to "No wheelchair", any reasonable UX would consider that a return to the default state. But ours doesn't. Instead it puts it in a 3rd visual state that… just shouldn't exist at all.

Apologies for not spec'ing it clearly.

[*] Technically, Ride Quality has 4 states and behaves visually as if there are 5. But the argument is the same.

**Comment by flibbertigibbet** [comment]
So are you saying that if an option is _changed_ to a selection that happens to be the default, it should revert to the unselected option text in the top-level modal? Because that's not what was in the prototype; each selection has it's own text. If a user picks a thing, we showed what they picked, not reverting to appear as if there were no selection.

**Comment by lederer** [comment]
> So are you saying that if an option is changed to a selection that happens to be the default, it should revert to the unselected option text in the top-level modal? 

Yes, I am saying that.

> Because that's not what was in the prototype; each selection has it's own text. If a user picks a thing, we show what they picked, not reverting to appear as if there were no selection.

I'm looking at the prototype. I don't agree it shows that. But I'll wholeheartedly grant it's ambiguous enough to have created confusion. I thought it was clear at the time. It was not. I've offered two apologies in this thread for not spec'ing it clearly.
